src/app/customer/checkoutSuggestion/BoltCheckoutSuggestion.tsx

```diff
@@ -0,0 +1,94 @@
+import { CheckoutSelectors, CustomerInitializeOptions, CustomerRequestOptions, ExecutePaymentProviderCheckoutOptions } from '@bigcommerce/checkout-sdk';
+import { noop } from 'lodash';
+import React, { memo, useEffect, useState, FunctionComponent, SyntheticEvent } from 'react';
+
+import { TranslatedString } from '../../locale';
+import { Button, ButtonVariant } from '../../ui/button';
+import { IconBolt } from '../../ui/icon';
+
+export interface BoltCheckoutSuggestionProps {
+    isExecutingPaymentProviderCheckout: boolean;
+    methodId: string;
+    deinitializeCustomer(options: CustomerRequestOptions): Promise<CheckoutSelectors>;
+    executePaymentProviderCheckout(options: ExecutePaymentProviderCheckoutOptions): Promise<CheckoutSelectors>;
+    initializeCustomer(options: CustomerInitializeOptions): Promise<CheckoutSelectors>;
+    onUnhandledError?(error: Error): void;
+}
+
+const BoltCheckoutSuggestion: FunctionComponent<BoltCheckoutSuggestionProps> = ({
+    isExecutingPaymentProviderCheckout,
+    methodId,
+    deinitializeCustomer,
+    executePaymentProviderCheckout,
+    initializeCustomer,
+    onUnhandledError = noop,
+}) => {
+    const [ showSuggestion, setShowSuggestion ] = useState<boolean>(false);
+
+    useEffect(() => {
+        deinitializeCustomer({ methodId });
+
+        try {
+            initializeCustomer({
+                methodId,
+                [methodId]: {
+                    onInit: (hasBoltAccount: boolean) => {
+                        setShowSuggestion(hasBoltAccount);
+                    },
+                },
+            });
+        } catch (error) {
+            onUnhandledError(error);
+        }
+
+        return () => {
+            deinitializeCustomer({ methodId });
+        };
+    }, [initializeCustomer, deinitializeCustomer, methodId, onUnhandledError]);
+
+    if (!showSuggestion) {
+        return null;
+    }
+
+    const handleContainerClick = (event: SyntheticEvent)  => {
```

Comment:
You can import `preventDefault` function from `'../common/dom'` and use it instead.

Reply:
I cant use it here, because I need to stop event propagation to prevent opening checkout step on click inside suggestion block.

Event prevent default didn't work to handle the case.

I can create another helper function an import it here. Is it will be ok?

Reply:
I see what you mean, you want to `stopPropagation` only. Yes, you can do that and import it here, seems generic enough to have one.